torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -359,7 +359,7 @@ void Reducer::mark_variable_ready(VariableIndex index) {
   // Check if this was the final gradient for this bucket.
   if (--replica.pending == 0) {
     // Prescale bucket contents to turn the global sum into the global average.
-    replica.contents.div_(process_group_->getSize());
+    replica.contents.mul_(1.0/process_group_->getSize());
```

Comment:
iter.common_dtype will be the type of replica.contents, which is floating point, so the division will automatically be converted  to multiplication.

Reply:
Ohhhh, thanks for pointing that out. The comment in the kernel was a bit confusing. Closing this PR.